engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1203,17 +1209,33 @@ private AllocateResponse sendContainerAskToRM(List<ContainerRequest> containerRe
       amRmClient.releaseAssignedContainer(containerId);
     }
 
+    return amRmClient.allocate(0);
+  }
+
+  private void processContainerStopRequests()
```

Comment:
Method name was misleading, it was sending a request to both RM and NM. Now it is clearly separated and this also addresses @sanjaypujare's code review comments.